src/nl/hannahsten/texifyidea/inspections/latex/probablebugs/LatexSuspiciousSectionFormattingInspection.kt

```diff
@@ -0,0 +1,80 @@
+package nl.hannahsten.texifyidea.inspections.latex.probablebugs
+
+import com.intellij.codeInspection.InspectionManager
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiFile
+import com.intellij.webSymbols.references.WebSymbolReferenceProvider.Companion.startOffsetIn
+import nl.hannahsten.texifyidea.inspections.InsightGroup
+import nl.hannahsten.texifyidea.inspections.TexifyInspectionBase
+import nl.hannahsten.texifyidea.psi.LatexCommands
+import nl.hannahsten.texifyidea.psi.LatexPsiHelper
+import nl.hannahsten.texifyidea.psi.LatexRequiredParam
+import nl.hannahsten.texifyidea.util.containsAny
+import nl.hannahsten.texifyidea.util.files.commandsInFile
+import nl.hannahsten.texifyidea.util.firstChildOfType
+import nl.hannahsten.texifyidea.util.magic.CommandMagic
+import nl.hannahsten.texifyidea.util.requiredParameter
+
+open class LatexSuspiciousSectionFormattingInspection : TexifyInspectionBase() {
+
+    override val inspectionGroup = InsightGroup.LATEX
+
+    override fun getDisplayName() = "Suspicious formatting in the required argument of a sectioning command"
+
+    override val inspectionId = "SuspiciousSectionFormatting"
+
+    override fun inspectFile(file: PsiFile, manager: InspectionManager, isOntheFly: Boolean): List<ProblemDescriptor> {
+        return file.commandsInFile()
+            .asSequence()
+            .filter { it.name in CommandMagic.sectionMarkers }
+            .filter { it.optionalParameterMap.isEmpty() }
+            .filter { it.requiredParameter(0)?.containsAny(formatting) == true }
+            .map { psiElement ->
+                val requiredParam = psiElement.firstChildOfType(LatexRequiredParam::class)
+                // Plus 1 for the opening brace.
+                val startOffset = requiredParam?.startOffsetIn(psiElement)?.plus(1) ?: 0
+                // Minus 2 for the braces surrounding the parameter.
+                val endOffset = requiredParam?.textLength?.minus(2)?.plus(startOffset) ?: psiElement.textLength
+                manager.createProblemDescriptor(
+                    psiElement,
+                    TextRange(startOffset, endOffset),
+                    "Suspicious formatting in ${psiElement.name}",
+                    ProblemHighlightType.WARNING,
+                    isOntheFly,
+                    AddOptionalArgumentQuickFix()
+                )
+            }
+            .toList()
+    }
+
+    class AddOptionalArgumentQuickFix : LocalQuickFix {
+
+        override fun getFamilyName(): String {
+            return "Add optional argument without formatting"
```

Comment:
Maybe make it clearer from a functional perspective what is happening, e.g. "Fix formatting in table of contents", if that's the only thing the optional argument is doing?